new version & changelog (#1583)
ScilifelabDataCentre/dds_web: CHANGELOG.rst

Changelog
==========

.. _2.9.0:

> 2.8.2 - 2024-12-13

Reviewer: ```suggestion
2.9.0 - 2024-12-18
```

~~~~~~~~~~~~~~~~~~~~~~~
- Bugs fixed:
    - Change the error raised upon attempt to download data after a password reset.
    - Fix the dds ls --tree command.
- New features:

>     - Update backend Dockerfile to pin a fixed version of mariadb-client.

Reviewer 2: Is this a new feature, maybe it fits better in bug fixes

    - Add documentation to new client feature of 'Upload' or 'Download' added to end of delivery directory name.
    - Modify the monitor usage command to send warning to the affected unit as well as Data Centre.

> Logging:

Reviewer: ```suggestion
- Logging:
```

    - Removed exception for invalid token to simplify logging.

>     - Logging: Add which user name reset password.

Reviewer: ```suggestion
    - Username logged when password is reset.
```

> - New features:
>     - Add documentation to new client feature of 'Upload' or 'Download' added to end of delivery directory name.
>     - Modify the monitor usage command to send warning to the affected unit as well as Data Centre.
> - Bugs fixed:
>     - Change the error raised upon attempt to download data after a password reset.
>     - Fix the dds ls --tree command.
>     - Update backend Dockerfile to pin a fixed version of mariadb-client.
> - Logging:
>     - Removed exception for invalid token to simplify logging.
>     - Username logged when password is reset.
>     - Filter out the MaintenanceModeException from the logs.

Reviewer: Think of this section from the users perspective: If you were the user, how would you best understand the changes. For me, this would be the following. 
```suggestion
- New features:
    - Documentation updated regarding 'Upload' or 'Download' at the end of staging directory 
    - The unit's contact email is now also notified when the unit is approaching the allocated data quota
- Bugs fixed:
    - Change in error if there is a download attempt after the password has been reset
    - The 'dds ls --tree' command works again
    - The mariadb-client version is now pinned in the dockerfile
- Logging:
    - The exception raised when there's an invalid token has been removed to simplify logging
    - Username logged when password is reset
    - MaintenanceModeException is filtered out from the logs
```

Reviewer: Also, don't forget to add a row regarding the CVE's.

.. _2.9.0:

> 2.9.0 - 2024-12-16

Reviewer: The new version is being deployed on the 18th.

Reviewer: ```suggestion
2.9.0 - 2024-12-18
```
